api/app/Actions/Profile/UpdateProfileAction.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Profile;
+
+use App\Exceptions\Profile\ProfileNotFoundException;
+use App\Repositories\User\UserRepository;
+use Illuminate\Auth\AuthenticationException;
+use Illuminate\Support\Facades\Auth;
+
+final class UpdateProfileAction
```

Comment:
I would change it to `UpdateUserAction` (and related classes and names) because we should not use different names for the same entity.